src/preact/base-element.js

```diff
@@ -254,17 +373,19 @@ export class PreactBaseElement extends BaseElement {
 
   /** @override */
   buildCallback() {
-    const Ctor = this.constructor;
+    const Ctor = /** @type {typeof PreactBaseElement} */ (
+      /** @type {unknown} */ (this.constructor)
+    );
 
     this.observer = new MutationObserver((rs) => this.checkMutations_(rs));
-    const props = Ctor['props'];
+    const {props} = Ctor;
```

Comment:
Is this safe? I thought we used quote props sometimes where we want to avoid mangling, but you'd know better than I

Reply:
Yep! It was potentially unsafe in the CC days, but with `terser` we only mangle private properties. Any reason we still do this is purely habit at this point